File: 01952.py
M, N = map(int, input().split())
snail = [[0] * (N + 2)] + [[0] + [1] * N + [0] for i in range(M)] + [[0] * (N + 2)]

x = 1; move_x = [1, 0, -1, 0]
y = 1; move_y = [0, 1, 0, -1]
dir = 0; cnt = 0; s = M * N
while True:
    if snail[y][x]:
        snail[y][x] = 0
        s -= 1
        if s == 0:
            break

        x += move_x[dir]
        y += move_y[dir]
    else:
        x -= move_x[dir]
        y -= move_y[dir]
        cnt += 1
        dir = (dir + 1) % 4
        x += move_x[dir]
        y += move_y[dir]

print(cnt)

File: 01952.cpp
#include <iostream>
using namespace std;

int main() {
	int snail[101][101] = { 0, };
	int M, N; cin >> M >> N;

	for (int i = 1; i <= M; i++)
		for (int j = 1; j <= N; j++)
			snail[i][j] = 1;

	int x = 1, y = 1, cnt = 0;
	while (1) {
		// 오른쪽으로 이동
		while (snail[y][x]) {
			snail[y][x] = 0;
			x++;
		}
		x--;
		if (!snail[y + 1][x]) break;
		else {
			cnt++;
			y++;
		}

		// 밑으로 이동
		while (snail[y][x]) {
			snail[y][x] = 0;
			y++;
		}
		y--;
		if (!snail[y][x - 1]) break;
		else {
			cnt++;
			x--;
		}

		// 왼쪽으로 이동
		while (snail[y][x]) {
			snail[y][x] = 0;
			x--;
		}
		x++;
		if (!snail[y - 1][x]) break;
		else {
			cnt++;
			y--;
		}

		// 위로 이동
		while (snail[y][x]) {
			snail[y][x] = 0;
			y--;
		}
		y++;
		if (!snail[y][x + 1]) break;
		else {
			cnt++;
			x++;
		}
	}

	cout << cnt << endl;
}
